library/kotlin/src/io/envoyproxy/envoymobile/RequestHeaders.kt

```diff
@@ -0,0 +1,55 @@
+
+package io.envoyproxy.envoymobile
+
+/*
+ * Headers representing an outbound request.
+ */
+class RequestHeaders: Headers {
+  /**
+   * Internal constructor used by builders.
+   *
+   * @param headers: Headers to set.
+   */
+  internal constructor(headers: Map<String, List<String>>) : super(headers)
+
+  /**
+   * Method for the request.
+   */
+  val method = RequestMethod.enumValue(value(":method")?.first()!!)
```

Comment:
In Swift these are computed and are only computed when they're accessed by the consumer. Is it possible to do that here @buildbreaker?

Reply:
yeah we have to use the `lazy` keyword: https://stackoverflow.com/questions/36623177/kotlin-property-initialization-using-by-lazy-vs-lateinit

Reply:
thanks, done!